src/test/java/com/wikia/webdriver/elements/mercury/pages/discussions/PageWithPosts.java

```diff
@@ -4,10 +4,42 @@
 import com.wikia.webdriver.elements.mercury.components.discussions.common.Post;
 import com.wikia.webdriver.elements.mercury.components.discussions.common.PostEntity;
 import com.wikia.webdriver.elements.mercury.components.discussions.common.SignInToFollowModalDialog;
+import com.wikia.webdriver.elements.mercury.components.discussions.desktop.*;
+import com.wikia.webdriver.elements.mercury.components.discussions.mobile.DiscussionsHeader;
+import com.wikia.webdriver.elements.mercury.components.discussions.mobile.FiltersPopOver;
 import com.wikia.webdriver.pageobjectsfactory.pageobject.WikiBasePageObject;
+import lombok.Getter;
 
 public abstract class PageWithPosts extends WikiBasePageObject {
 
+  public abstract SignInToFollowModalDialog getSignInDialog();
+
+  public abstract PageWithPosts open();
+
+  @Getter(lazy = true)
+  private final SortingFilterDesktop sortingFilterDesktop = new SortingFilterDesktop();
+
+  @Getter(lazy = true)
+  private final DiscussionsHeader discussionsHeader = new DiscussionsHeader();
+
+  @Getter(lazy = true)
+  private final FiltersPopOver filtersPopOver = new FiltersPopOver();
+
+  @Getter(lazy = true)
+  private final Promoting promoting = new Promoting();
+
+  @Getter(lazy = true)
+  private final CommunityBadge communityBadge = new CommunityBadge();
+
+  @Getter(lazy = true)
+  private final HeroUnit heroUnit = new HeroUnit();
+
+  @Getter(lazy = true)
+  private final Moderation moderation = new Moderation();
+
+  @Getter(lazy = true)
+  private final Post post = new Post();
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Move this variable to comply with Java Code Conventions. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar-s4/coding_rules#rule_key=squid%3AS1213)